extensions/mock-credentials-verifier/src/test/java/org/eclipse/dataspaceconnector/mvd/MockCredentialsVerifierTest.java

```diff
@@ -14,33 +14,47 @@
 
 package org.eclipse.dataspaceconnector.mvd;
 
-import org.eclipse.dataspaceconnector.iam.did.spi.key.PublicKeyWrapper;
+import org.eclipse.dataspaceconnector.iam.did.spi.document.DidDocument;
+import org.eclipse.dataspaceconnector.iam.did.spi.document.Service;
 import org.eclipse.dataspaceconnector.spi.EdcException;
 import org.eclipse.dataspaceconnector.spi.monitor.ConsoleMonitor;
 import org.eclipse.dataspaceconnector.spi.result.Result;
 import org.junit.jupiter.api.Test;
 
+import java.util.List;
 import java.util.Map;
+import java.util.UUID;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
-import static org.mockito.Mockito.mock;
+import static org.eclipse.dataspaceconnector.identityhub.credentials.VerifiableCredentialsJwtService.VERIFIABLE_CREDENTIALS_KEY;
 
 class MockCredentialsVerifierTest {
+    private static final String CREDENTIAL_SUBJECT_KEY = "credentialSubject";
     MockCredentialsVerifier verifier = new MockCredentialsVerifier(new ConsoleMonitor());
-    PublicKeyWrapper wrapper = mock(PublicKeyWrapper.class);
 
     @Test
     void verifyCredentials() {
-        Result<Map<String, Object>> actual = verifier.verifyCredentials("http://dummy.site/foo?region=us&tier=GOLD", wrapper);
+        Result<Map<String, Object>> actual = getVerifiedCredentials("http://dummy.site/foo?region=us&tier=GOLD");
         assertThat(actual.succeeded()).isTrue();
         assertThat(actual.getContent())
+                .extracting(c -> c.values().stream().findFirst().get())
+                .extracting(VERIFIABLE_CREDENTIALS_KEY)
+                .extracting(CREDENTIAL_SUBJECT_KEY)
                 .isEqualTo(Map.of("region", "us", "tier", "GOLD"));
     }
 
     @Test
     void verifyCredentials_failsOnMalformedUrl() {
-        assertThatThrownBy(() -> verifier.verifyCredentials("malformed_url", wrapper))
+        assertThatThrownBy(() -> getVerifiedCredentials("malformed_url"))
```

Comment:
nit: one more test can be added where DID document is empty. For example, it doesn't have identityHubUrl. That will cause a Result.failure response.